src/StatusQ/Buttons/StatusButton.qml

```diff
@@ -0,0 +1,165 @@
+import QtQuick 2.14
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Components 0.1
+
+
+Rectangle {
+    id: statusButton
+
+    enum Size {
+        Small,
+        Large
+    }
+
+    enum Type {
+        Normal,
+        Danger,
+        WithoutBackground,
+        DangerWithoutBackground
+    }
+
+    property bool loading: false
+
+    property int size: StatusButton.Size.Large
+    property int type: StatusButton.Type.Normal
+
+    property alias text: label.text
+    property alias font: label.font
+
+    property real leftPadding: size === StatusButton.Size.Large ? 24 : 12 // by design
+    property real rightPadding: size === StatusButton.Size.Large ? 24 : 12 // by design
+    property real topPadding: size === StatusButton.Size.Large ? 11 : 10 // by design
+    property real bottomPadding: size === StatusButton.Size.Large ? 11 : 10 // by design
+
+    signal pressed(var mouse)
```

Comment:
Any reason we aren't calling this "clicked" ?

I personally am a fan of keeping as much of "native" component feeling around. Users that will use `StatusButton` might not expect an `onPressed` signal

Reply:
Every "clickable" components have signals `pressed` `released `clicked` and `pressAndHold`. So we don't have `pressed` state of button. `clicked` is pressed+released. If you want i can add it. Its just my `legacy`

Reply:
I guess the question is: What is most likely used/expected by users?

So far we've been using `onClicked` on all buttons in Status Desktop. If you think `onPressed` is rather the way to go, then that's fine by me. We'll just have to touch many places in the app to make sure nothing breaks.

We could also just have both `clicked` and `pressed`. If we want to be consistent, we should probably have all of the events mentioned above, because, it's a button after all (one of the reasons I've inherited from `Button` instead of doing a custom `Rectangle`)

Reply:
So no problem, i will forward signals from `MouseArea`